initial attempt

sudesnac opened the pull request:
for probtrackx2 with exclusion mask

Commit:
initial attempt

Commit:
fslmaths

akhanf committed:
Merge branch 'master' into exclude_roi

Commit:
modified

Commit:
Merge branch 'exclude_roi' of https://github.com/akhanf/basalfb-diffparc into exclude_roi

Commit:
added fslmaths to combine excrois

akhanf commented on Snakefile:
```diff
@@ -25,207 +25,229 @@ f.close()
 rule all:
-    input: 
-        clusters = expand('diffparc/clustering/group_space-{template}_seed-{seed}_hemi-{hemi}_method-spectralcosine_k-{k}_cluslabels.nii.gz',seed=seeds,hemi=hemis,template=config['template'],k=range(2,config['max_k']+1))
+	input: 
```
your editor seems to be using tabs for indents instead of spaces -- spaces (4spaces for an indent) is the python standard so should stick with that -- what editor are you using? should be able to force it to do this..

sudesnac replied:
I am using Microsoft Visual Studio  - yes noticed that and was trying to fix... will search if there's any shortcut / setting...

akhanf commented on Snakefile:
```diff
@@ -25,207 +25,229 @@ f.close()
+		seed_nii = 'diffparc/sub-{subject}/masks/seed_{seed}_{hemi}.nii.gz'
+	shell: 'cp -v {input} {output}'
+
+rule import_excrois_subject:
```
since this rule is effectively taking all the excrois and combining into a single image, you don't want to have a {excroi} wildcard (since that would invoke a job for each excroi).. instead you want to use the `expand()` function to loop over excrois in the inputs

akhanf replied:
I've made a commit just now with that suggestion, have a look and let me know what you think

akhanf replied:
for the shell command, an easier strategy could be something like:
```
fslmerge -t {output.combined_4d} {input.excroi_nii} &&
fslmaths {output.combined_4d} -Tmax {output.com_excrois}
```

sudesnac replied:
> since this rule is effectively taking all the excrois and combining into a single image, you don't want to have a {excroi} wildcard (since that would invoke a job for each excroi).. instead you want to use the expand() function to loop over excrois in the inputs

is there a way to use expand() and just take the excrois and not the subjects? I tried to use expand() but it takes all the subjects as well (which it should) but then I am concerned if we merge all excrois over the subjects would it not interfere with protract? shouldn't the --avoid {exclusion mask} be applied for each subject in their native space?

akhanf replied:
yes, the way to do exactly that is to use the `allow_missing=True` option, that basically lets you skip other wildcards in the expand statement  (i.e. it will keep {subject} as a wildcard)

akhanf replied:
a couple simple examples to illustrate: 
if we had: 
```
subjects=['1','2']
excroi=['a','b']
```
This one:
```
expand('{subject}_{excroi}.nii.gz', subject=subjects,excroi=excrois)
```
would evaluate to:
`['1_a.nii.gz', '1_b.nii.gz', '2_a.nii.gz', '2_b.nii.gz']`

but this:
```
expand('{subject}_{excroi}.nii.gz',excroi=excrois, allow_missing=True)
```
would evaluate to:
`['{subject}_a.nii.gz', '{subject}_b.nii.gz']`

akhanf committed:
modification to input/output/log to use expand()

haven't touched the shell: section yet, but this is the proper set of inputs/outputs for a rule that is going to act on all the excrois at once

Commit:
added shell command - fslmerge and fslmaths to combine excrois

Commit:
modification

Commit:
slight modification for importing excrois

Commit:
minor modification

Commit:
minor modification

Commit:
singularity container modified for mrtrix

sudesnac committed:
update group

Commit:
modify to fix ambiguous rule

Commit:
minor changes to excroi

Commit:
changes and dag file

Commit:
rule graph

Commit:
organize dag file

Commit:
visualize results

sudesnac committed:
modification

Commit:
slight change

Commit:
resolve merge conflict

akhanf committed:
Merge branch 'master' into exclude_roi

akhanf committed:
Delete cluster_visualize-checkpoint.ipynb
